### ISSUE TITLE
Class placement on images in the wrong place

### ISSUE THREAD

**Issue by rossahall** [WordPress/gutenberg#9345]
Behaviour:
When an image block is added to the editor a number of attributes are either missing or applied incorrectly. These limit customisation of individual images in art direction and for accessibility purposes.

Specifically:

"Image Title Attribute" cannot be set for the specific instance of the image
"Image CSS Class" cannot be set for the specific instance of the image for art direction or basic responsive purposes (e.g. to distinguish between a portrait, square or landscape image)
"Link rel" cannot be set
"Link CSS" cannot be set

In it's place we have a "custom class" attribute which is applied to a "figure" tag surrounding the image set, which is not backwards compatible to pre-HTML 5 browsers.

**Comment by youknowriad** [comment]
Thanks for opening the issue.

> Browser support

Gutenberg like WordPress support these browsers https://make.wordpress.org/core/handbook/best-practices/browser-support/ and all these images support the figure tag properly.


> Classnames

If you have a className on the image wrapper, you can style it exactly as if it were on the image.

`.selector > img`

> Link attributes

Notice that Gutenberg is extensible and you can build any advanced image block you want with any customizable attribute. The image block is built to support the common and popular use-cases, and the popular use-cases, the rest is plugin territory. Can you explain a bit more how do you plan to use these attributes, it's not very clear. Do you have examples of markup?

**Comment by rossahall** [comment]
Thanks, but this does not address the issue raised. Individual classes cannot be set for images / links etc.

Example code (abbreviated)

<a href="somelink" class="a specific style"><img srcset etc class="a specific image style"></a>

<a href="some other link" class ="a different style"><img srcset etc class="a different image style"></a>

<a href="yet another link" class="a specific style"><img srcset etc class="a different image style"></a>

(Etc)

Easily achieved in V4 and backwards compatible into pre- <figure> supporting browsers.

Custom code to recreate functionality current present in Wordpress isn't acceptable.

**Comment by youknowriad** [comment]
I'm not sure I understand what's stopping you from styling the link in images like this `.my-custom-class > a`?

**Comment by rossahall** [comment]
Because I would have to create 4 custom styles to encapsulate the different combinations of link and image styles.

That's just with 2 separate styles for images and links. Add another style and the problem increases exponentially.

The functionality to assign custom classes to links and images currently exists and is incredibly flexible. Seems the "fix" is less than flexible and more "tech solution" than designer's.

**Comment by youknowriad** [comment]
The possibility to dd rel and a custom className to image links has been added recently.
